nft-provider-aggregator/contracts/NFTProviderAggregator.cdc

```diff
@@ -30,154 +31,170 @@ import NonFungibleToken from "./NonFungibleToken.cdc"
 
     /// Private paths for Aggregator{SupplierFactory}, Aggregator{SupplierAccess},
     /// and Aggregator{NonFungibleToken.Provider} capabilities
-    pub let SupplierFactoryPrivatePath: PrivatePath
-    pub let SupplierAccessPrivatePath: PrivatePath
-    pub let AggregatedProviderPrivatePath: PrivatePath
-    
+    /// Note: These paths are kept for backward compatibility purposes
+    /// (for use with convertPrivateToStoragePath and getPrivateCapPathFromStoragePath)
+    access(all) let SupplierFactoryPrivatePath: PrivatePath
+    access(all) let SupplierAccessPrivatePath: PrivatePath
+    access(all) let AggregatedProviderPrivatePath: PrivatePath
+
     /// Interface that a supplier would commonly use for publicly exposing
     /// the Supplier resource's getter functions through a public capability
     ///
-    pub resource interface SupplierPublic {
-        pub fun getAggregatorUUID(): UInt64
-        pub fun getSupplierAddedCollectionUUIDs(): [UInt64]
-        pub fun getCollectionUUIDs(): [UInt64]
-        pub fun getIDs(): [UInt64]
+    access(all) resource interface SupplierPublic {
+        access(all) view fun getAggregatorUUID(): UInt64
+        access(all) view fun getSupplierAddedCollectionUUIDs(): [UInt64]
+        access(all) view fun getCollectionUUIDs(): [UInt64]
+        access(all) fun getIDs(): [UInt64]
     }
 
     /// Interface that a manager would commonly use for exposing the Aggregator resource's
     /// createSupplier function to each designated supplier through a private capability
     ///
-    pub resource interface SupplierFactory {
-        pub fun createSupplier(): @Supplier
+    access(all) resource interface SupplierFactory {
+        access(Operate) fun createSupplier(): @Supplier
     }
 
     /// Interface used by a manager for exposing core Aggregator resource's functions to the Supplier
     /// resource through a private capability passed to the Supplier at resource creation time
     ///
-    pub resource interface SupplierAccess {
-        access(contract) fun addNFTProviderCapability(
-            nftProviderCapability: Capability<
-            &AnyResource{NonFungibleToken.Provider, NonFungibleToken.CollectionPublic}>
-            ): UInt64
-        access(contract) fun removeNFTProviderCapability(collectionUUID: UInt64)
-        access(contract) fun getIDs(): [UInt64]
-        access(contract) fun getCollectionUUIDs(): [UInt64]
+    access(all) resource interface SupplierAccess {
+        access(Operate) fun addNFTWithdrawCapability(
+            _ cap: Capability<auth(NonFungibleToken.Withdraw) &{NonFungibleToken.Collection}>): UInt64
+        access(Operate) fun removeNFTWithdrawCapability(collectionUUID: UInt64)
+        access(all) fun getIDs(): [UInt64]
+        access(all) view fun getCollectionUUIDs(): [UInt64]
     }
 
+    /// Entitlement that grants the ability to operate the Escrow Collection
+    ///
+    access(all) entitlement Operate
```

Comment:
New `Operate` entitlement used for access control both for `Aggregator` and `Supplier` methods